### ISSUE TITLE
Error only while debugging in VSCode

### ISSUE THREAD

**Issue by sophia-wright-blue** [julia-vscode/julia-vscode#3349]
Hello - I'm able to run a Julia script in VSCode, but when I set breakpoints and run in Debug mode, the same code throws an exception.

I have described the issue in detail with screenshots and the exact error here:

https://github.com/CliMA/Oceananigans.jl/issues/3171

the issue is due to some functions returning the value of variables (boundary condition values more specifically) as `nothing` , which is not an error, and as I mentioned before, when I run the script, it runs fine.

I hope the thread in the linked issue provides more information.

I'd greatly appreciate any help figuring out how to resolve this issue so I can use the debugger. Thank you!

**Comment by sophia-wright-blue** [comment]
update to this issue:

the author of the `KernelAbstractions ` library has added a comment in the issue linked above that this sounds like a fault in the debugger - it'd be super helpful if we could get some feedback on resolving this issue - thank you

**Comment by sophia-wright-blue** [comment]
I've shared a minimal reproducer here in case it helps resolve this issue:

https://github.com/CliMA/Oceananigans.jl/issues/3171#issuecomment-1624346600

**Comment by pfitzseb** [comment]
This doesn't seem to be VS Code specific, but rather looks like a general issue with JuliaInterpreter:
```
julia> @interpret mul2_kernel(dev, 64)(A, ndrange=size(A))
ERROR: invalid lookup expr $(Expr(:aliasscope))
Stacktrace:
  [1] error(::String, ::Expr)
    @ Base ./error.jl:44
  [2] lookup_expr(frame::Frame, e::Expr)
    @ JuliaInterpreter ~/.julia/packages/JuliaInterpreter/ktCFr/src/interpret.jl:25
  [3] eval_rhs(recurse::Any, frame::Frame, node::Expr)
    @ JuliaInterpreter ~/.julia/packages/JuliaInterpreter/ktCFr/src/interpret.jl:409
  [4] step_expr!(recurse::Any, frame::Frame, node::Any, istoplevel::Bool)
    @ JuliaInterpreter ~/.julia/packages/JuliaInterpreter/ktCFr/src/interpret.jl:546
  [5] step_expr!(recurse::Any, frame::Frame, istoplevel::Bool)
    @ JuliaInterpreter ~/.julia/packages/JuliaInterpreter/ktCFr/src/interpret.jl:596
  [6] finish!(recurse::Any, frame::Frame, istoplevel::Bool)
    @ JuliaInterpreter ~/.julia/packages/JuliaInterpreter/ktCFr/src/commands.jl:14
  [7] finish_and_return!(recurse::Any, frame::Frame, istoplevel::Bool)
    @ JuliaInterpreter ~/.julia/packages/JuliaInterpreter/ktCFr/src/commands.jl:30
  [8] evaluate_call_recurse!(recurse::Any, frame::Frame, call_expr::Expr; enter_generated::Bool)
    @ JuliaInterpreter ~/.julia/packages/JuliaInterpreter/ktCFr/src/interpret.jl:260
  [9] evaluate_call_recurse!
    @ ~/.julia/packages/JuliaInterpreter/ktCFr/src/interpret.jl:220 [inlined]
 [10] eval_rhs(recurse::Any, frame::Frame, node::Expr)
    @ JuliaInterpreter ~/.julia/packages/JuliaInterpreter/ktCFr/src/interpret.jl:393
 [11] step_expr!(recurse::Any, frame::Frame, node::Any, istoplevel::Bool)
    @ JuliaInterpreter ~/.julia/packages/JuliaInterpreter/ktCFr/src/interpret.jl:546
 [12] step_expr!(recurse::Any, frame::Frame, istoplevel::Bool)
    @ JuliaInterpreter ~/.julia/packages/JuliaInterpreter/ktCFr/src/interpret.jl:596
 [13] finish!(recurse::Any, frame::Frame, istoplevel::Bool)
    @ JuliaInterpreter ~/.julia/packages/JuliaInterpreter/ktCFr/src/commands.jl:14
 [14] finish_and_return!(recurse::Any, frame::Frame, istoplevel::Bool)
    @ JuliaInterpreter ~/.julia/packages/JuliaInterpreter/ktCFr/src/commands.jl:30
 [15] evaluate_call_recurse!(recurse::Any, frame::Frame, call_expr::Expr; enter_generated::Bool)
    @ JuliaInterpreter ~/.julia/packages/JuliaInterpreter/ktCFr/src/interpret.jl:260
 [16] evaluate_call_recurse!
    @ ~/.julia/packages/JuliaInterpreter/ktCFr/src/interpret.jl:220 [inlined]
 [17] eval_rhs(recurse::Any, frame::Frame, node::Expr)
    @ JuliaInterpreter ~/.julia/packages/JuliaInterpreter/ktCFr/src/interpret.jl:393
 [18] step_expr!(recurse::Any, frame::Frame, node::Any, istoplevel::Bool)
    @ JuliaInterpreter ~/.julia/packages/JuliaInterpreter/ktCFr/src/interpret.jl:546
 [19] step_expr!(recurse::Any, frame::Frame, istoplevel::Bool)
    @ JuliaInterpreter ~/.julia/packages/JuliaInterpreter/ktCFr/src/interpret.jl:596
 [20] finish!(recurse::Any, frame::Frame, istoplevel::Bool)
    @ JuliaInterpreter ~/.julia/packages/JuliaInterpreter/ktCFr/src/commands.jl:14
 [21] finish_and_return!(recurse::Any, frame::Frame, istoplevel::Bool)
    @ JuliaInterpreter ~/.julia/packages/JuliaInterpreter/ktCFr/src/commands.jl:30
 [22] evaluate_call_recurse!(recurse::Any, frame::Frame, call_expr::Expr; enter_generated::Bool)
    @ JuliaInterpreter ~/.julia/packages/JuliaInterpreter/ktCFr/src/interpret.jl:260
 [23] evaluate_call_recurse!
    @ ~/.julia/packages/JuliaInterpreter/ktCFr/src/interpret.jl:220 [inlined]
 [24] eval_rhs(recurse::Any, frame::Frame, node::Expr)
    @ JuliaInterpreter ~/.julia/packages/JuliaInterpreter/ktCFr/src/interpret.jl:393
 [25] step_expr!(recurse::Any, frame::Frame, node::Any, istoplevel::Bool)
    @ JuliaInterpreter ~/.julia/packages/JuliaInterpreter/ktCFr/src/interpret.jl:546
 [26] step_expr!(recurse::Any, frame::Frame, istoplevel::Bool)
    @ JuliaInterpreter ~/.julia/packages/JuliaInterpreter/ktCFr/src/interpret.jl:596
 [27] finish!(recurse::Any, frame::Frame, istoplevel::Bool)
    @ JuliaInterpreter ~/.julia/packages/JuliaInterpreter/ktCFr/src/commands.jl:14
 [28] finish_and_return!(recurse::Any, frame::Frame, istoplevel::Bool)
    @ JuliaInterpreter ~/.julia/packages/JuliaInterpreter/ktCFr/src/commands.jl:30
 [29] evaluate_call_recurse!(recurse::Any, frame::Frame, call_expr::Expr; enter_generated::Bool)
    @ JuliaInterpreter ~/.julia/packages/JuliaInterpreter/ktCFr/src/interpret.jl:260
 [30] evaluate_call_recurse!
    @ ~/.julia/packages/JuliaInterpreter/ktCFr/src/interpret.jl:220 [inlined]
 [31] eval_rhs(recurse::Any, frame::Frame, node::Expr)
    @ JuliaInterpreter ~/.julia/packages/JuliaInterpreter/ktCFr/src/interpret.jl:393
 [32] step_expr!(recurse::Any, frame::Frame, node::Any, istoplevel::Bool)
    @ JuliaInterpreter ~/.julia/packages/JuliaInterpreter/ktCFr/src/interpret.jl:546
 [33] step_expr!(recurse::Any, frame::Frame, istoplevel::Bool)
    @ JuliaInterpreter ~/.julia/packages/JuliaInterpreter/ktCFr/src/interpret.jl:596
 [34] finish!(recurse::Any, frame::Frame, istoplevel::Bool)
    @ JuliaInterpreter ~/.julia/packages/JuliaInterpreter/ktCFr/src/commands.jl:14
 [35] finish_and_return!
    @ ~/.julia/packages/JuliaInterpreter/ktCFr/src/commands.jl:30 [inlined]
 [36] finish_and_return!
    @ ~/.julia/packages/JuliaInterpreter/ktCFr/src/commands.jl:34 [inlined]
 [37] finish_and_return!(frame::Frame)
    @ JuliaInterpreter ~/.julia/packages/JuliaInterpreter/ktCFr/src/commands.jl:34
 [38] top-level scope
    @ ~/.julia/packages/JuliaInterpreter/ktCFr/src/construct.jl:758
```

**Comment by sophia-wright-blue** [comment]
thank you so much for looking into this @pfitzseb - could you please advice me on the next steps to fix the issue - should I open an issue on the main Julia language repo: https://github.com/JuliaLang/julia

or is there some other repo that would be more suitable?

thank you again

**Comment by pfitzseb** [comment]
Try opening an issue [here](https://github.com/JuliaDebug/JuliaInterpreter.jl/issues).

**Comment by sophia-wright-blue** [comment]
thank you so much - the issue is linked above - hopefully, someone will respond soon!

**Comment by sophia-wright-blue** [comment]
hello @pfitzseb - just got feedback that the above issue in JuliaInterpreter.jl has been fixed and a new version released:

https://github.com/JuliaDebug/JuliaInterpreter.jl/issues/574#issuecomment-1674729615

https://github.com/JuliaDebug/JuliaInterpreter.jl/releases


will this julia-vscode repo have to be updated to use the new release?

**Comment by pfitzseb** [comment]
Yep. That'll likely take a bit.

**Comment by sophia-wright-blue** [comment]
ok - I'll keep track of the releases - thank you again for your help - looking forward to being able to use the debugger with Julia :)

**Comment by davidanthoff** [comment]
I think `master` should now have the latest build, so I'll close this for now.

**Comment by sophia-wright-blue** [comment]
hello @davidanthoff @pfitzseb - thank you for releasing a new version - I updated the Julia VSCode extension - I'm not sure if I'm doing something wrong, but I'm still getting the exact same error as before

it'd help me greatly if you could check if you're able to run the minimal reproducer that was giving you an error before - thank you

**Comment by davidanthoff** [comment]
Which version of the extension are you using? We only pushed the new build as a pre-release, so you wouldn't get it unless you opt-in to that.

**Comment by sophia-wright-blue** [comment]
I switched to the pre-release version - 1.49.1 - I'm getting the same error on that

**Comment by sophia-wright-blue** [comment]
hello @davidanthoff @pfitzseb - I updated the Julia-vscode extension to v1.50.2 and I'm able to debug the reproducer without any issues now - this is amazing!

it'd be great to know how you fixed the most recent issue - I will test this on the original library some more and close this issue - thank you so much for your help!

**Comment by pfitzseb** [comment]
Pretty sure I just didn't properly update to the latest JuliaInterpreter version before we released 1.49. https://github.com/julia-vscode/julia-vscode/commit/eb69f7425e1bd73e7efc2098f69d5bb22490185c#diff-4f39dd244658c05a1899ab917ad737bffcda785ff6d387029969d3f593526fb6 "fixed" that.

**Comment by pfitzseb** [comment]
Unfortunately that is expected behaviour. Run your code on a smaller grid if you can to cut down on the runtime, hopefully without impacting the behaviour you want to debug.
